[WIP] feat: intent에 따른 message는 FormGroup 컴포넌트에서 보여주도록 한다

chiabi opened the pull request:
- formInput 컴포넌트들에 중복되는 message를 제거하고 intent상태만 관리하도록 한다. message와 message의 intent는 FormGroup컴포넌트에서 사용하도록 한다.
- formInput 컴포넌트 중 disabled 상태 스타일링이 없거나 cursor:now-allowed가 적용되어 있지 않는 부분을 수정한다.
- FormGroup 컴포넌트에 helperText를 추가한다
- FormGroup, formInputs의 변경사항에 대한 문서를 추가한다

chiabi committed:
feat: intent에 따른 message는 FormGroup 컴포넌트에서 보여주도록 한다

- formInput 컴포넌트들에 중복되는 message를 제거하고 intent상태만 관리하도록 한다. message와 message의 intent는 FormGroup컴포넌트에서 사용하도록 한다.
- formInput 컴포넌트 중 disabled 상태 스타일링이 없거나 cursor:now-allowed가 적용되어 있지 않는 부분을 수정한다.
- FormGroup 컴포넌트에 helperText를 추가한다
- FormGroup, formInputs의 변경사항에 대한 문서를 추가한다

chiabi committed:
feat: ContorlGroup 스타일을 수정한다

- 필요없는 vertical, fill props를 삭제한다

chiabi committed:
feat: TagInput의 disabled 상태 스타일을 수정한다

chiabi committed:
refactor: redError 쓴 곳을 red500으로 수정한다

bae-unidev commented on src/core/GlobalStyle/index.tsx:
```diff
@@ -12,4 +12,7 @@ export const GlobalStyle = createGlobalStyle`
     padding: 0;
     margin: 0
   }
+  h1,h2,h3,h4,h5,h6 {
```
이거 왠지 우리 웹에 적용되면 스타일 깨질것 같은데??

bae-unidev commented on src/core/common/intent.ts:
```diff
@@ -0,0 +1,15 @@
+import classNames from 'classnames';
```
우리가 언제부터 classNames을 쓰기 시작했지?

styled-components로 합칠수 있으니 합치자!

bae-unidev commented on src/formControls/FormGroup/index.tsx:
```diff
@@ -1,61 +1,103 @@
-      <Container inline={inline} {...divAttributes} {...restProps}>
-        <Label labelColor={labelColor} htmlFor={htmlLabelFor}>
+      <Container isHorizontal={isHorizontal} {...divAttributes} {...restProps}>
+        <Label color={labelColor || gray800} htmlFor={htmlLabelFor} shouldAlignLabel={isHorizontal && hasHelperText}>
```
`labelColor || gray800`는 defaultProps로 처리하자~

bae-unidev commented on src/formInputs/Checkbox/index.tsx:
```diff
@@ -1,76 +1,59 @@
+          disabled={disabled}
+          {...inputAttributes}
+          {...restProps}
+          {...inputRef && {
```
이거 React.forwardRef로 구현해줘~

bae-unidev commented on src/formInputs/DatePicker/index.tsx:
```diff
@@ -255,6 +278,9 @@ export class DatePicker extends PureComponent<DatePickerProps, DatePickerState>
   };
 
   private handleClickDecreaseBtn = () => {
```
우리가 언제부터 약어를 썼지?

bae-unidev commented on src/formInputs/Input/index.tsx:
```diff
@@ -1,102 +1,77 @@
+        {...inputAttributes}
+        {...restProps}
+        {...inputRef && {
+          ref: inputRef,
```
forwardRef ㄲ

bae-unidev commented on src/formInputs/Select/index.tsx:
```diff
@@ -1,45 +1,56 @@
         value={value}
         color={value === '' ? gray500 : gray800}
-        {...selectAttributes}
+        fill={`${fill}`}
```
```suggestion
        fill={fill}
```

chiabi replied:
이거는 
`Warning: Received `true` for non-boolean attribute` 때문에 적용한 거

https://github.com/styled-components/styled-components/issues/1198

chiabi committed:
feat: classnames를 제거한다

Button, FormInputs에서 사용한 classNames를 제거한다

chiabi committed:
feat: Link가 a로 props를 전달하면서 생기는 경고 메시지 제거

chiabi committed:
refactor: formControl 관련해서 리팩토링한다

- GlobalStyle에서 기존 스타일에 영향을 미칠 수 있어서 heading tag 스타일 초기화한 부분을 다시 제거한다
- defaultProps에 정의할 수 있는 걸 뺀다
- 약어로 쓰인 메서드 이름을 수정한다

chiabi committed:
refactor: forwardRef로 리팩토링 한다

chiabi committed:
refactor: forwardRef를 사용한 formInputs을 functional component로 변경한다

ddalpange commented on docs/formInputs/TagInput.mdx:
```diff
@@ -44,3 +44,128 @@ import { TagInput } from '@class101/ui';
+      constructor(props) {
+        super(props);
+        this.state = {
+          value: ['01082210886','01082230886', '01082210886', '01082210886','01082210886','01082210886', '01082210886', '01082210886','01082210886','01082210886', '01082210886', '01082210886','01082210886','01082210886', '01082210886', '01082210886']
```
선생님 이거 제 개인정보거든요.
Lorem Ipsum으로 바꿔주세요.

chiabi replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 그래요

chiabi committed:
refactor: TagInput 예시를 lorem ipsum으로 변경한다

chiabi commented:
oui에서 진행하도록 하겠습니당
